python/instrumentation/openinference-instrumentation-bedrock/src/openinference/instrumentation/bedrock/__init__.py

```diff
@@ -158,7 +164,75 @@ def instrumented_response(*args: Any, **kwargs: Any) -> Dict[str, Any]:
                 span.set_attributes(dict(get_attributes_from_context()))
                 return response  # type: ignore
 
-        return instrumented_response
+        @wraps(wrapped_client.converse)
+        def instrumented_response_converse(*args: Any, **kwargs: Any) -> Dict[str, Any]:
+            if context_api.get_value(_SUPPRESS_INSTRUMENTATION_KEY):
+                return wrapped_client._unwrapped_converse(*args, **kwargs)  # type: ignore
+
+            with tracer.start_as_current_span("bedrock.converse") as span:
+                span.set_attribute(
+                    SpanAttributes.OPENINFERENCE_SPAN_KIND,
+                    OpenInferenceSpanKindValues.LLM.value,
+                )
+
+                if model_id := kwargs.get("modelId"):
+                    _set_span_attribute(span, SpanAttributes.LLM_MODEL_NAME, model_id)
+
+                if request_messages := kwargs.get("messages"):
+                    # Extract the most recent request message in message history list
+                    current_request_message = request_messages[-1]
+
+                    if request_content := current_request_message.pop("content", ""):
+                        # Currently only supports single text-based data
+                        # Future implementation can be extended to handle different media types
+                        if prompt := request_content[-1].get("text"):
+                            _set_span_attribute(span, SpanAttributes.INPUT_VALUE, prompt)
```

Comment:
nit: i'm not a pythonista is there a way to simplify this if nesting? if messages, if content, if prompt etc. nothing wrong with this just a preference - will leave it up to you and the python folks!